actors/builtin/miner/miner_actor.go

```diff
@@ -857,6 +861,59 @@ func (a Actor) DeclareFaultsRecovered(rt Runtime, params *DeclareFaultsRecovered
 	return nil
 }
 
+func (a Actor) processSkippedFaults(rt Runtime, st State, skipped *bitfield.BitField, deadline *DeadlineInfo,
+	epochReward abi.TokenAmount, qaPowerTotal abi.StoragePower) ([]*SectorOnChainInfo, abi.TokenAmount) {
+
+	empty, err := skipped.IsEmpty()
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "failed to check if skipped sectors is empty")
+	if empty {
+		return nil, abi.NewTokenAmount(0)
+	}
+
+	currEpoch := rt.CurrEpoch()
+	store := adt.AsStore(rt)
+	var skippedFaultSectors []*SectorOnChainInfo
+
+	deadlines, err := st.LoadDeadlines(adt.AsStore(rt))
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load deadlines")
+
+	// Check that the declared sectors are actually due at the deadline.
+	deadlineSectors := deadlines.Due[deadline.Index]
+	contains, err := abi.BitFieldContainsAll(deadlineSectors, skipped)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "could not check if skipped faults are in deadline")
+	if !contains {
+		rt.Abortf(exitcode.ErrIllegalArgument, "skipped faults contains sectors not due in deadline %d", deadline.Index)
+	}
+
+	// Find all skipped faults that have been labeled recovered
+	retractedRecoveries, err := bitfield.IntersectBitField(st.Recoveries, skipped)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to intersect sectors with recoveries")
+
+	// Ignore skipped faults that are already faults but have not been declared recovered
+	newFaults, err := bitfield.SubtractBitField(skipped, st.Faults)
```

Comment:
The highlighted line is only half the operation. The next line restores the retracted recoveries. It's `(S ∖ F) ∪ (S ∩ R)` instead of `S ∖ (F ∖ R)`. Since `R ⊆ F` they're equivalent, and we already have the right side. It's moot though since we need to track skipped faults and skipped recoveries separately.